pkg/jwt/parse.go

```diff
@@ -0,0 +1,44 @@
+package jwt
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/lestrrat-go/jwx/jwk"
+	"github.com/lestrrat-go/jwx/jwt"
+	"github.com/pkg/errors"
+)
+
+// rawJWTFromAuthHeader returns a raw jwt token from the auth header
+func rawJWTFromAuthHeader(r *http.Request) string {
+	auth := r.Header.Get("Authorization")
+	if auth == "" {
+		return ""
+	}
+
+	auth = strings.TrimPrefix(auth, "bearer ")
+	return strings.TrimPrefix(auth, "Bearer ")
+}
+
+// ParseTokenFromRequestWithJWKS Parses a jwt.Token from an Auth header, validating with JWKS Keys from an endpoint
+func ParseTokenFromRequestWithJWKS(r *http.Request, jwksEndpoint string) (jwt.Token, error) {
+	keyset, err := jwk.Fetch(r.Context(), jwksEndpoint)
+	if err != nil {
+		return nil, errors.Wrap(err, "ParseTokenFromHeaderWithJWKS():jwk.Fetch()")
+	}
+
+	// Disable "infer algorithm" so this always uses the algorithm specified at the jwks endpoint or fails
+	// (avoid something along the lines of https://auth0.com/blog/critical-vulnerabilities-in-json-web-token-libraries/)
+	token, err := jwt.ParseString(rawJWTFromAuthHeader(r), jwt.WithKeySet(keyset), jwt.InferAlgorithmFromKey(false))
```

Comment:
You never check if the return from rawJWTFromAuthHeader is an empty string. If it's an empty string, does it matter? If not, both comments can be dismissed.